aea/crypto/base.py

```diff
@@ -83,3 +85,49 @@ def dump(self, fp: BinaryIO) -> None:
         :param fp: the output file pointer. Must be set in binary mode (mode='wb')
         :return: None
         """
+
+
+class LedgerApi(ABC):
+    """Interface for ledger APIs."""
+
+    identifier = None  # type: Optional[str]
+
+    @abstractmethod
+    def get_balance(self, address: AddressLike) -> int:
+        """
+        Get the balance of a given account.
+
+        This usually takes the form of a web request to be waited synchronously.
+
+        :param address: the address.
+        :return: the balance.
+        """
+
+    @abstractmethod
+    def send_transaction(self,
+                         crypto_object: Crypto,
+                         destination_address: AddressLike,
+                         amount: int,
+                         tx_fee: int
+                         ) -> Optional[str]:
+        """
+        Submit a transaction to the ledger.
+
+        :param crypto_object: the crypto object associated to the payer.
+        :param destination_address: the destination address of the payee.
+        :param amount: the amount of wealth to be transferred.
+        :param tx_fee: the transaction fee.
+        :return: the transaction digest, or None.
+        """
+
+    # TODO?
+    # def get_transaction_receipt(self, tx_digest: str) -> Dict:
+
+    @abstractmethod
+    def is_transaction_settled(self, tx_digest: str) -> bool:
```

Comment:
Add optional param which is the amount, so you can check that the tx references the correct amount.

Reply:
optional payee too